features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Scopes/EditScope.jsx

```diff
@@ -0,0 +1,231 @@
+     * @memberof Scopes
+     */
+    addScope() {
+        const restApi = new Api();
+        const updatedScope = this.state.apiScope;
+        const { intl } = this.props;
+        const { api } = this.props;
+        api.scopes = api.scopes.map((scope) => {
+            if (scope.name === updatedScope.name) {
+                return updatedScope;
+            } else {
+                return scope;
+            }
+        });
+        const { history } = this.props;
+        // eslint-disable-next-line no-underscore-dangle
+        const promisedApiUpdate = restApi.update(api._data);
+        promisedApiUpdate.then((response) => {
+            if (response.status !== 200) {
+                Alert.info(intl.formatMessage({
+                    id: 'Apis.Details.Scopes.CreateScope.something.went.wrong.while.updating.the.scope',
+                    defaultMessage: 'Something went wrong while updating the scope',
+                }));
+                return;
+            }
+            Alert.info(intl.formatMessage({
+                id: 'Apis.Details.Scopes.CreateScope.scope.updated.successfully',
+                defaultMessage: 'Scope updated successfully',
+            }));
+            const redirectURL = '/apis/' + api.id + '/scopes/';
+            history.push(redirectURL);
+        });
+    }
+
+    /**
+     * Handle api scope addition event
+     * @param {any} event Button Click event
+     * @memberof Scopes
+     */
+    handleInputs(event) {
+        if (Array.isArray(event)) {
+            const { apiScope } = this.state;
+            apiScope.bindings.values = event;
+            this.setState({
+                apiScope,
+            });
+        } else {
+            const input = event.target;
+            const { apiScope } = this.state;
+            apiScope[input.id] = input.value;
+            this.setState({
+                apiScope,
+            });
+        }
+    }
+
+    render() {
+        const { classes } = this.props;
+        const url = `/apis/${this.props.api.id}/scopes`;
+        return (
+            <Grid container>
+                <Typography
+                    className={classes.headline}
+                    gutterBottom
+                    variant='headline'
+                    component='h2'
+                >
+                    <FormattedMessage
+                        id='Apis.Details.Scopes.EditScope.update.scope'
+                        defaultMessage='Update Scope'
+                    />
+                </Typography>
+                <Grid item lg={5} className={classes.topics}>
+                    <APIPropertyField name='Name'>
+                        <TextField
+                            disabled
+                            fullWidth
+                            id='name'
+                            type='text'
+                            name='name'
+                            margin='normal'
+                            value={this.state.apiScope.name}
```

Comment:
destructure `this.state.apiScope` and use as `apiScope.name`, `apiScope.description`, `apiScope.bindings.values`